### PR TITLE
rustfmt: fix on darwin

### PR DESCRIPTION
**Pull request by PuercoPop**
###### Description of changes
Without this patch rustfmt installs successfully but upon running it it throws the following error

```
  $ rustfmt
  dyld[68147]: Library not loaded: @rpath/librustc_driver-c577224f5690b349.dylib
    Referenced from: <no uuid> /nix/store/w46inc6cad6xg2kw09v9n629iqb7aqrw-rustfmt-1.69.0/bin/rustfmt
    Reason: tried: '/nix/store/ih18sy1kcbl5kdgq2l8l1lm70ai6aybj-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/librustc_driver-c577224f5690b349.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/nix/store/ih18sy1kcbl5kdgq2l8l1lm70ai6aybj-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/librustc_driver-c577224f5690b349.dylib' (no such file), '/nix/store/ih18sy1kcbl5kdgq2l8l1lm70ai6aybj-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/librustc_driver-c577224f5690b349.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/nix/store/ih18sy1kcbl5kdgq2l8l1lm70ai6aybj-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/librustc_driver-c577224f5690b349.dylib' (no such file), '/usr/local/lib/librustc_driver-c577224f5690b349.dylib' (no such file), '/usr/lib/librustc_driver-c577224f5690b349.dylib' (no such file, not in dyld cache)
  Abort trap: 6
```

Reading this [rust-issue] it seems that we have to link against rustc_driver. The [clippy.nix] expression already does something similar

Of the 4 executables found in the result of building rustfmt only rustfmt and git-rustfmt needed to be linked. The other worked without linking to rustc_driver.

[rust-issue]: https://github.com/rust-lang/rust/pull/105609
[clippy.nix]: https://github.com/NixOS/nixpkgs/blob/c8cf570dae9b41f30395b71f9b432418b4ff0ebc/pkgs/development/compilers/rust/clippy.nix#L27-L36

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PuercoPop** [comment]
The editorconfig check is failing, I can't find the error message in the web UI. I tried reproducing the failure locally but it doesn't seem to complain:

```
cd ~/src/nixpkgs
$ nix-shell -p editorconfig-checker

[nix-shell:~/src/nixpkgs]$ editorconfig-checker -disable-indent-size pkgs/development/compilers/rust/rustfmt.nix 

[nix-shell:~/src/nixpkgs]$ echo $?
0
```

**Comment by winterqt** [comment]
Blame GitHub -- I re-ran it, and it passed.

File: pkgs/development/compilers/rust/rustfmt.nix [hunk 1]
```diff
@@ -15,6 +15,11 @@ rustPlatform.buildRustPackage rec {
     rustPlatform.rust.rustc.llvm
   ] ++ lib.optional stdenv.isDarwin Security;
 
+  preFixup = lib.optionalString stdenv.isDarwin ''
```
Comment (winterqt) [inline pkgs/development/compilers/rust/rustfmt.nix:18]: Might be worth adding my comment from Clippy's derivation here, or just reference it.

**Review by winterqt — CHANGES_REQUESTED** [review]
Commit message needs to be in proper form, e.g. `rustfmt: fix on darwin`.

**Comment by winterqt** [comment]
I'd also argue that the commit message can be shortened significantly, as we already explain why this workaround is needed in the Clippy derivation/commits, but maybe I'm nitpicking too much :)

File: pkgs/development/compilers/rust/rustfmt.nix (hunk 1 above)
Comment (PuercoPop) [reply pkgs/development/compilers/rust/rustfmt.nix:24]: I agree that it should be commented. I was unsure on what was the best way to do so. I've copied the comment verbatim.

**Comment by PuercoPop** [comment]
I've updated the commit header line as requested and removed some of the specific details of the commit message. Let me know what you think

**Comment by winterqt** [comment]
Just changed the commit message to remove the extraneous Darwin mention.

**Review by winterqt — APPROVED** [review]
Tested, thanks!

**Comment by winterqt** [comment]
Skipping OfBorg because I've already tested this extensively, and it seems to be very backed up right now.

**Pull request merged**